lib/std/posix.zig

```diff
@@ -4331,6 +4373,46 @@ pub fn fstatat(dirfd: fd_t, pathname: []const u8, flags: u32) FStatAtError!Stat
     }
 }
 
+// MKDEV in the kernel.
+inline fn makedevLinux(major: u32, minor: u32) dev_t {
+    return (major << 20) | minor;
+}
+
+fn fstatatStatxZ(dirfd: fd_t, pathname: [*:0]const u8, stat_buf: *Stat, flags: u32) usize {
+    var statx = mem.zeroes(linux.Statx);
+    const rc = linux.statx(dirfd, pathname, flags | AT.NO_AUTOMOUNT, linux.STATX_BASIC_STATS, &statx);
+    const rcs: isize = @bitCast(rc);
+
+    if (rcs > -4096) return rc;
+
+    stat_buf.* = .{
+        .dev = makedevLinux(statx.dev_major, statx.dev_minor),
+        .rdev = makedevLinux(statx.rdev_major, statx.rdev_minor),
+        .ino = statx.ino,
+        .mode = statx.mode,
+        .nlink = statx.nlink,
+        .uid = statx.uid,
+        .gid = statx.gid,
+        .atim = .{
+            .tv_sec = statx.atime.tv_sec,
+            .tv_nsec = statx.atime.tv_nsec,
+        },
+        .mtim = .{
+            .tv_sec = statx.mtime.tv_sec,
+            .tv_nsec = statx.mtime.tv_nsec,
+        },
+        .ctim = .{
+            .tv_sec = statx.ctime.tv_sec,
+            .tv_nsec = statx.ctime.tv_nsec,
+        },
+        .size = @as(i64, @intCast(statx.size)),
+        .blocks = @as(i64, @intCast(statx.blocks)),
+        .blksize = @as(i32, @intCast(statx.blksize)),
```

Comment:
Not completely sure about the theoretical safety of these. Hard to imagine it'd be a problem in practice, though.

Reply:
Added a comment that points this out in the latest version of the commit.